bbl_screen-patch/patches/printerui/qml/x1plus/DDS.js

```diff
@@ -45,3 +46,21 @@ _DdsListener.gotDdsEvent.connect(function(topic, dstr) {
         }
     }
 });
+_DdsListener.gotDdsEvent.connect(function(topic, dstr) {
+    const datum = JSON.parse(dstr);
+    console.log("x1p",dstr);
+    if (topic == "device/report/info") {
+        if (datum["command"] != "get_version")
+            return;
+        _setVersions(datum['module']);
+    } else if (topic == "device/report/print") {
+        if (datum["command"] == "push_status") {
+            if (gcodeAction() != datum["print_gcode_action"]) {
+                _setGcodeAction(datum["print_gcode_action"]);
+            }
+        } else if (datum["gpio"] == "action" ){
+            var action_code = parseInt(datum["action_code"]);
+            X1Plus.GpioKeys.handleAction(action_code,datum["param"]);
+        }
+    }
+});
```

Comment:
duplicates most of the above?  (also, match on `command`, not `gpio`?)

Reply:
This should be changed to _setGpioAction() to match the rest of the format, and yes that should be matching "command" to "gpio_action", not "gpio" to "action". That was a mistake I didn't catch. This is the DDS message format I ended up with: 

>"command": "gpio_action",
>"action_code": action_code,
>"param": params,